scripts/build

```diff
@@ -6,6 +6,21 @@ source $(dirname $0)/lib/version
 
 cd $(dirname $0)/
 
-./build-engine $6
-./build-routeagent $6
-./build-globalnet $6
+# Outer loop causes getopts to not-quit on unknown arguments
```

Comment:
It might be easier to understand if we move `set -e` after the arguments are parsed, instead of wrapping `getopts` in another loop.

Reply:
Why won't we want to quit on unknown params? It's an error.

I know the Makefile does some magic to declare the targets, but honestly I think we can live with them being defined statically and more correctly (i.e. not sending parameters where they're not needed).

Reply:
You're saying you want to get rid of the "make targets are automatically searched for as scripts in scripts/ dir" system? I'm not sure I'm on board with that one.

Reply:
I do want to get rid of that part of the Makefiles, to avoid the issues with, *e.g.* `bin/subctl` in the operator project. However that’s a separate discussion, which shouldn’t affect this PR. I think this PR should preserve the current behaviour, so ignoring errors in arguments is acceptable IMO.
I stand by my comment on the outer loop, it shouldn’t be necessary.

Reply:
We could preserve it for now but I would expect it to be removed later on since it's quite error prone to do it so dynamically, and surely there's a better way to write the makefile :)

Reply:
Okay, understood and agreed @skitt and @mkolesnik. I think the current version gets rid of the loop y'all didn't like, moves to getopt, but doesn't scope-creep into additional changes to the makefile. Thanks for the good feedback!